block/manager.go

```diff
@@ -170,36 +181,63 @@ func (m *Manager) GetFraudProofOutChan() chan *abci.FraudProof {
 }
 
 // AggregationLoop is responsible for aggregating transactions into rollup-blocks.
-func (m *Manager) AggregationLoop(ctx context.Context) {
-	initialHeight := uint64(m.genesis.InitialHeight)
-	height := m.store.Height()
-	var delay time.Duration
+func (m *Manager) AggregationLoop(ctx context.Context, ingress chan []byte, progressive bool) {
+	if !progressive {
+		initialHeight := uint64(m.genesis.InitialHeight)
+		height := m.store.Height()
+		var delay time.Duration
+
+		// TODO(tzdybal): double-check when https://github.com/celestiaorg/rollmint/issues/699 is resolved
+		if height < initialHeight {
+			delay = time.Until(m.genesis.GenesisTime)
+		} else {
+			delay = time.Until(m.lastState.LastBlockTime.Add(m.conf.BlockTime))
+		}
 
-	// TODO(tzdybal): double-check when https://github.com/celestiaorg/rollmint/issues/699 is resolved
-	if height < initialHeight {
-		delay = time.Until(m.genesis.GenesisTime)
-	} else {
-		delay = time.Until(m.lastState.LastBlockTime.Add(m.conf.BlockTime))
-	}
+		if delay > 0 {
+			m.logger.Info("Waiting to produce block", "delay", delay)
+			time.Sleep(delay)
+		}
 
-	if delay > 0 {
-		m.logger.Info("Waiting to produce block", "delay", delay)
-		time.Sleep(delay)
-	}
+		timer := time.NewTimer(0)
 
-	timer := time.NewTimer(0)
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-timer.C:
+				start := time.Now()
+				err := m.publishBlock(ctx, false)
+				if err != nil {
+					m.logger.Error("error while publishing block", "error", err)
+				}
+				timer.Reset(m.getRemainingSleep(start))
+			}
+		}
+	} else {
+		m.logger.Info("Starting Progressive Aggregation Loop")
```

Comment:
remove the `Info` and `Debug` statements as they are not very useful.

Reply:
deleted some, kept some others. let me know if you think i should delete them all